src/Stores/Promises/selectors.js

```diff
@@ -23,9 +23,22 @@ export const promiseSubjectsSelector: (
       const links = subjectsPromises
         ? subjectsPromises.filter(sP => sP.promise === promise.id)
         : [];
-      const subs = links.map(
-        link => (subjects ? subjects.find(s => link.subject === s.id) : [])
-      );
+      const subs = links.map(link => {
+        if (!subjects) {
+          // TODO trigger loading of subjects instead of throwing error
```

Comment:
This something you want to do in this PR?

Reply:
hadn't planned on it right now.
`PromisesContainer` makes sure everything is loaded

Reply:
maybe create an issue just so its not forgotten?

Reply:
thinking more about this.
The selector should just return `null` if the data hasn't been loaded.

It's the consumers responsibility to ensure that data has been loaded

Reply:
https://github.com/veritus/veritus-web/issues/81 <- me coming to that conclusion